casper/src/main/resources/PoS.rho

```diff
@@ -1,180 +1,165 @@
+    posRevAddressCh,
+    eitherCh,
+    revVaultCh,
+    getUser,
+    getCurrentUserAddress,
+    getCurrentUserVault,
+    deposit in {
+
+  registryLookup!(`rho:id:j6trahbxycumerwpr5qype7j43b7eqh8auebwsa9nn68if47gswh73`, *eitherCh) |
+  registryLookup!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *revVaultCh) |
+
+  for(@(_, Either) <- eitherCh;
+      @(_, RevVault) <- revVaultCh) {
+    new posRevAddressCh in {
+  //TODO - set the user properly when running PoSTest.rho
+  //    getCurrentUserAddress!(*posRevAddressCh)|
+
+      revAddressOps!("fromPublicKey", "0000000000000000000000000000000000000000000000000000000000000000".hexToBytes(), *posRevAddressCh) |
+
+      for(@posRevAddress <- posRevAddressCh) {
+
+        contract MakePoS(@minimumBond, @maximumBond, @initialBonds, posReturnCh) = {
+          new bondsCh,
+              PoS in {
+            posReturnCh!(*PoS) |
+            bondsCh!(initialBonds) |
+            getCurrentUserVault!(Nil) |
+
+            contract PoS (@"getBonds", returnCh) = {
+              for(@bonds <- bondsCh) {
+                bondsCh!(bonds) | returnCh!(bonds)
               }
-
-              // TODO: Validate that the associated wallet comes from the same mint as the purse
-              // TODO: Can we assume at this point that [[@pk]] is a valid public key?
-              | contract PoS (@"bond", @bondAmount, return) = {
-
-                new bondResultCh,
-                  isBonded,
-                  validateBondAmount,
-                  validateBondingRate,
-                  validatePublicKey,
-                  updateBonds,
-                  parametersCh in {
-
-                  @Either!(
-                    "compose",
-                    bondAmount,
-                    [*validateBondAmount,
-                    *validateBondingRate,
-                    *validatePublicKey,
-                    *deposit,
-                    *updateBonds],
-                    *return
-                  )
-
-                  | contract isBonded (@userPk, return) = {
-                    for(@bonds <- bondsCh) {
-                      bondsCh!(bonds) | return!(bonds.contains(userPk))
-                    }
-                  }
-
-                  | contract validateBondAmount (@bondAmount, return) = {
-                    if (bondAmount < minimumBond) {
-                      return!((false, "Bond less than minimum!"))
-                    } else if (bondAmount > maximumBond) {
-                      return!((false, "Bond greater than maximum!"))
-                    } else {
-                      return!((true, bondAmount))
-                    }
-                  }
-
-                  // TODO: figure out rate-limiting of bonding
-                  | contract validateBondingRate (@bondAmount, return) = {
-                    return!((true, bondAmount))
-                  }
-
-                  | contract validatePublicKey (@bondAmount, return) = {
```

Comment:
Did you remove this check because key verification is performed by `RevVault`?

Reply:
This contract checks if the key is already bonded. This check exists in my implementation too. See line 65